idaes/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/hetero_reactions.py

```diff
@@ -0,0 +1,517 @@
+                                    is_reaction_parameter_block)
+import idaes.logger as idaeslog
+
+# Some more information about this module
+__author__ = "Chinedu Okoli"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("HeteroReactionParameterBlock")
+class ReactionParameterData(ReactionParameterBlock):
+    """
+    Property Parameter Block Class
+
+    Contains parameters and indexing sets associated with properties for
+    superheated steam.
+
+    """
+    # Create Class ConfigBlock
+    CONFIG = ConfigBlock()
+    CONFIG.declare("gas_property_package", ConfigValue(
+            description="Reference to associated PropertyPackageParameter "
+                        "object for the gas phase.",
+            domain=is_physical_parameter_block))
+    CONFIG.declare("solid_property_package", ConfigValue(
+            description="Reference to associated PropertyPackageParameter "
+                        "object for the solid phase.",
+            domain=is_physical_parameter_block))
+    CONFIG.declare("default_arguments", ConfigBlock(
+            description="Default arguments to use with Property Package",
+            implicit=True))
+
+    def build(self):
+        '''
+        Callable method for Block construction.
+        '''
+        super(ReactionParameterBlock, self).build()
+
+        self._reaction_block_class = ReactionBlock
+
+        # Create Phase objects
+        self.Vap = VaporPhase()
+        self.Sol = SolidPhase()
+
+        # Create Component objects
+        self.N2 = Component()
+        self.O2 = Component()
+        self.CO2 = Component()
+        self.H2O = Component()
+        self.Fe2O3 = Component()
+        self.Fe3O4 = Component()
+        self.Al2O3 = Component()
+
+        # Component list subsets
+        self.gas_component_list = Set(initialize=['O2', 'N2', 'CO2', 'H2O'])
+        self.sol_component_list = Set(initialize=['Fe2O3', 'Fe3O4', 'Al2O3'])
+
+        # Reaction Index
+        self.rate_reaction_idx = Set(initialize=["R1"])
+
+        # Gas Constant
+        self.gas_const = Param(within=PositiveReals,
+                               mutable=False,
+                               default=8.314459848e-3,
+                               doc='Gas Constant [kJ/mol.K]')
+
+        # Reaction Stoichiometry
+        self.rate_reaction_stoichiometry = {("R1", "Vap", "O2"): -1,
+                                            ("R1", "Vap", "N2"): 0,
+                                            ("R1", "Vap", "CO2"): 0,
+                                            ("R1", "Vap", "H2O"): 0,
+                                            ("R1", "Sol", "Fe2O3"): 6,
+                                            ("R1", "Sol", "Fe3O4"): -4,
+                                            ("R1", "Sol", "Al2O3"): 0}
+
+        # Reaction stoichiometric coefficient
+        self.rxn_stoich_coeff = Param(self.rate_reaction_idx,
```

Comment:
Do you need this `Param` component? Why not just refer to the dict you just created?